File: sdks/csharp/src/NovaBilling/Invoices/Requests/ListInvoicesRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record ListInvoicesRequest
{
    [JsonIgnore]
    public string? Status { get; set; }

    [JsonIgnore]
    public string? CustomerId { get; set; }

    [JsonIgnore]
    public string? DateFrom { get; set; }

    [JsonIgnore]
    public string? DateTo { get; set; }

    [JsonIgnore]
    public double? Page { get; set; }

    [JsonIgnore]
    public double? Limit { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/ruby/lib/novabilling/invoices/types/list_invoices_request.rb
# frozen_string_literal: true

module Novabilling
  module Invoices
    module Types
      class ListInvoicesRequest < Internal::Types::Model
        field :status, -> { String }, optional: true, nullable: false
        field :customer_id, -> { String }, optional: true, nullable: false, api_name: "customerId"
        field :date_from, -> { String }, optional: true, nullable: false, api_name: "dateFrom"
        field :date_to, -> { String }, optional: true, nullable: false, api_name: "dateTo"
        field :page, -> { Integer }, optional: true, nullable: false
        field :limit, -> { Integer }, optional: true, nullable: false
      end
    end
  end
end
